### PR TITLE
Seth graphql

### PR DESCRIPTION
**Pull request by SethPerna**
Is this a fix or a feature?
- [ ] Bug fix
- [x] Feature
- [ ] Refactor
 
Describe the feature - fix - refactor:
Create GraphQL API endpoints for create and index. formats phone and ensure US valid phone number.

How was this tested?
- [ ] open index.html
- [x] localhost
- [ ] terminal

Where should the reviewer start - test:
Where ever your heart desires

Additional comments:
The lord giveth, he can taketh away

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -11,4 +11,26 @@ class User < ApplicationRecord
...
+     end
+   formatted.gsub!(";", " x")
+   formatted
+ end
+
+ private
+
+ def normalize_phone
+   self.phone_number = Phonelib.parse(phone_number).full_e164.presence
+ end
```
Comment (EMielke76) [inline app/models/user.rb:35]: This was super cool to see, thanks for walking me through how you built this out!

**Review by EMielke76 — COMMENTED** [review]
Looks good Seth! Good working through it alongside you!

File: app/graphql/mutations/create_user.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  argument :phone, String, required: true
+  argument :bio, String, required: true
+  argument :location, String, required: true
+  argument :donor_status, Integer, required: true
+
+  field :user, Types::UserType, null: false
+  field :errors, [String], null: false
+
+  def resolve(name:, email:, phone:, bio:, location:, donor_status:)
+    user = User.new(name: name, email: email, phone: phone, bio: bio, location: location, donor_status: donor_status)
```
Comment (dara-rockwell) [inline app/graphql/mutations/create_user.rb:13]: ⛏️ Consider putting each of the attributes onto a new line for easier grokkability.

File: app/models/user.rb [hunk 3]
```diff
@@ -11,4 +11,26 @@ class User < ApplicationRecord
 
   enum active_status: {active: 0, inactive: 1}
   enum donor_status: {donor: 0, recipient: 1}
+  before_save :normalize_phone
```
Comment (dara-rockwell) [inline app/models/user.rb:14]: ⛏️ Consider using `before_validations` instead, so phone numbers are normalized before `.valid?` is evaluated.

**Pull request merged**